File: vm/tests/all/interopComplex/references/testMain.js
//references.js

var llFile = Polyglot.evalFile("llvm", "references.so");

c = new llFile.MyClass();
c.i = 3;
ref = c.getRef();
ref = 4;

if(c.i==4) {
	console.log("OK");
} else {
	console.log("Error: 0 of 1 passed. c.i should be 4, but is ", c.i);
}

File: vm/tests/all/interopComplex/references/testMain.cpp
//reference testing

#include<stdio.h>
#include<polyglot.h>


int main() {
	PolyglotFile* llFile = Polyglot.evalFile("llvm", "references.so");

	MyClass* c = new llFile.LLVM_MyClass();
	c->i=3;
	int& ref = c->getRef();
	ref = 4;
	if(c->i==4) {
		printf("OK\n");
	} else {
		printf("Error: 0 of 1 passed. c->i should be 4, but is %i\n", c->i);
	}
	delete c;
	free(llFile);
}

